beets/library.py

```diff
@@ -1068,8 +1071,27 @@ def art_destination(self, image, item_dir=None):
                                      replacements=self._db.replacements)
         subpath = bytestring_path(subpath)
 
+        # Rectify the image extension based on its magic bytes. Mostly relevant
+        # for png with jpg extension.
+        try:
+            real_ext = imghdr.what(syspath(image))
+        except IOError:
+            real_ext = None
```

Comment:
It might be a good idea to use a whitelist here, along the lines of the MIME type whitelist in the fetchart plugin. That is, we could say that we'll change the extension only when we see JPEG and PNG data; for anything else, for instance if `imghdr` reports "tiff" or something else weird, we'll just leave the extension alone.